letsencrypt/client.py

```diff
@@ -125,6 +130,21 @@ def register(config, account_storage, tos_cb=None):
 
     return acc, acme
 
+def perform_registration(acme, config):
+    """
+    Actually register new account, trying repeatedly if there are email
+    problems
+
+    :returns: the same value as acme.register
+    """
+    try:
+        return acme.register(messages.NewRegistration.from_data(email=config.email))
+    except messages.Error, e:
+        if "MX record" in repr(e):
+            config.namespace.email = display_ops.get_email(more=True, invalid=True)
+            return perform_registration(acme, config)
```

Comment:
I don't really care if you fix this here and in `get_email` or not, but I think we should avoid recursion in Python since it doesn't support tail call elimination. See Guido's [blog](http://neopythonic.blogspot.com.au/2009/04/final-words-on-tail-calls.html) for more info on the issue.